src/plugin.js

```diff
@@ -231,18 +260,33 @@ const onPlayerReady = (player) => {
  * to you; if not, remove the wait for "ready"!
  *
  * @function eme
+ * @param    {Object} [player]
+ *           A player object.
  * @param    {Object} [options={}]
  *           An object of options left to the plugin author to define.
  */
-const eme = function(options = {}) {
-  this.eme.options = options;
-
-  this.ready(() => onPlayerReady(this));
+const eme = function(player, options = {}) {
+  player.ready(() => onPlayerReady(player));
+
+  player.eme = {
+    initializeMediaKeys(emeOptions = {}) {
+      const mergedEmeOptions = videojs.mergeOptions(
+        player.currentSource(),
```

Comment:
If you don't want to mutate any of these objects, then `videojs.mergeOptions({}, player.currentSource(), options, emeOptions);`

Reply:
The first arg shouldn't be mutated since the props are getting accumulated into a new object https://github.com/videojs/video.js/blob/master/src/js/utils/merge-options.js#L25

Reply:
I believe the state of the eme plugin is carrying over after player dispose somehow, I saw this when updating the tests.

Reply:
hmm, interesting

Reply:
manually setting `player.eme` may be breaking this https://github.com/videojs/video.js/blob/master/src/js/plugin.js#L304?